src/input/AnalysisAttributes.h

```diff
@@ -25,7 +27,7 @@
 using namespace tinyxml2;
 class AnalysisAttributes {
   public:
-    int * faceBound;
+    std::vector<std::pair<int, int>> faceBound;
```

Comment:
What is faceBound exactly? Maybe we could self-document this by using a vector of structs  instead of this approach?